Consul.Test/AgentTest.cs

```diff
@@ -479,18 +476,32 @@ public async Task Agent_NodeMaintenance()
         [Fact]
         public async Task Agent_Monitor()
         {
-            var logs = await _client.Agent.Monitor(LogLevel.Trace);
-            var counter = 0;
-            var successToken = new CancellationTokenSource();
-            var failTask = Task.Delay(5000, successToken.Token).ContinueWith(x => Assert.True(false, "Failed to finish reading logs in time"));
-            foreach (var line in logs)
+            using (var logs = await _client.Agent.Monitor(LogLevel.Trace))
             {
-                Assert.False(string.IsNullOrEmpty(await line));
-                counter++;
-                if (counter > 5) { break; }
+                var counter = 0;
+                var successToken = new CancellationTokenSource();
+                var timeoutTask = Task.Delay(TimeSpan.FromMinutes(1), successToken.Token).ContinueWith(x => Assert.True(false, "Failed to finish reading logs in time"));
+                var logsTask = Task.Run(async () =>
+                {
+                    // to get some logs
+                    await _client.Agent.Self();
+
+                    foreach (var line in logs)
+                    {
+                        // Make a request each time so we get more logs
+                        await _client.Agent.Self();
+                        Assert.False(string.IsNullOrEmpty(await line));
+                        counter++;
+                        if (counter > 5)
+                        {
+                            successToken.Cancel();
+                            break;
+                        }
+                    }
+                });
+
+                await await Task.WhenAny(new[] { timeoutTask, logsTask });
```

Comment:
Is `await await` on purpose here? (this is not a rhetorical question!)

Reply:
No it is not a typo 😄 
- `Task.WhenAny` creates a new task that returns one of the provided tasks
- first `await` gives as the task that completed first
- second `await` awaits the completed task which either does nothing or throws an exception (if the task has failed with an exception) 


Previously that code didn't work at all, because the exception from timeout assertion was never propagated to the main thread.